src/parser/ad_parser.js

```diff
@@ -27,29 +29,42 @@ export function parseAd(adElement) {
 
     parserUtils.copyNodeAttribute('id', adElement, adTypeElement);
     parserUtils.copyNodeAttribute('sequence', adElement, adTypeElement);
-
     if (adTypeElement.nodeName === 'Wrapper') {
-      return parseWrapper(adTypeElement);
+      return parseWrapper(adTypeElement, emit);
     } else if (adTypeElement.nodeName === 'InLine') {
-      return parseInLine(adTypeElement);
+      return parseInLine(adTypeElement, emit);
     }
   }
 }
 
 /**
- * Parses an Inline element.
- * @param  {Object} inLineElement - The VAST Inline element to parse.
+ * Parses an Inline
+ * @param  {Object} ad Element - The VAST Inline element to parse.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ * @emits  VASTParser#VAST-warning
+ * @return {Ad}
+ */
+function parseInLine(adElement, emit) {
+  return parseAdElement(adElement, emit);
+}
+
+/**
+ * Parses an ad type (Inline or Wrapper)
+ * @param  {Object} adTypeElement - The VAST Inline or Wrapper element to parse.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ * @emits  VASTParser#VAST-warning
  * @return {Ad}
  */
-function parseInLine(inLineElement) {
-  const childNodes = inLineElement.childNodes;
+function parseAdElement(adTypeElement, emit) {
+  parserVerification.verifyRequiredValues(adTypeElement, emit);
```

Comment:
you should do the check for no `emit` here, so you don't have to call this function